src/opni_nats/nats_wrapper.py

```diff
@@ -89,6 +90,40 @@ async def publish(self, nats_subject: str, payload_df):
         """
         await self.nc.publish(nats_subject, payload_df)
 
+    async def request(self, nats_subject: str, payload, timeout=1):
+        return await self.nc.request(nats_subject, payload, timeout=timeout)
+
+
+async def nats_request(nw, payload_queue):
+    while True:
+        payload = await payload_queue.get()
+        if payload is None:
+            continue
+        payload_df = pd.read_json(payload, dtype={"_id": object})
+
+        try:
+            response = await nw.request(
+                "logs",
+                payload_df.to_json().encode(),
+                timeout=1,
+            )
+            response = response.data.decode()
+        except ErrTimeout:
+            response = ""
+        print(response)
+
+
+async def nats_reply(nw):
+    async def receive_and_reply(msg):
+        reply_subject = msg.reply
+        if msg.data:
+            reply_message = b"NO"
+        else:
+            reply_message = b"YES"
+        await nw.publish(reply_subject, reply_message)
+
+    await nw.subscribe("logs", subscribe_handler=receive_and_reply)
```

Comment:
why is this outside the `NatsWrapper` class?

Reply:
can you add example usage in the main() method of this script to demonstrate how one would use your request-reply?
the current main demos how to use nats wrapper - just add to it. 

Take note that the functions `nats_subscriber` and `nats_publisher` will never actually be used in downstream scripts

Reply:
this is the example usage. I will add it to main.